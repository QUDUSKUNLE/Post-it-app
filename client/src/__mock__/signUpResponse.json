{
    "message": "Registration successful and verification email sent to your email",
    "response": [
        "https://post-it-app.firebaseio.com/users/FHeyW95bhLgLtpOTfFSN8fscPxD2/-KwyuT0Wa4EV7eqEpuuP",
        "https://post-it-app.firebaseio.com/Group/general/member/FHeyW95bhLgLtpOTfFSN8fscPxD2/-KwyuT0aSaR91R9eamqj",
        "FHeyW95bhLgLtpOTfFSN8fscPxD2"
    ]
}