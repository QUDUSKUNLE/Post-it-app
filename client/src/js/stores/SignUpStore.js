import { EventEmitter } from 'events';
import AppDispatcher from '../dispatcher/AppDispatcher.js';
import {
  SIGN_UP_SUCCESS, SIGN_UP_ERROR
} from '../constants/ActionConstants.js';

/**
 * Signout Store, it hold user's state, listen to signin Actions
 * @class SignUpStore
 */
class SignUpStore extends EventEmitter {
  /**
   * @constructor
   */
  constructor() {
    super();
    this.signUpMessage = {};
    this.signUpUser = this.signUpUser.bind(this);
    this.handleSignUpActions = this.handleSignUpActions.bind(this);
  }

  signUpUser() {
    return this.signUpMessage;
  }
  /**
   * Receives actions and update the stores accordingly
   * @method handleSignUpActions
   * @param {object} action - Action type and data
   * @return {null} -
   */
  handleSignUpActions(action) {
    switch (action.type) {
      case SIGN_UP_SUCCESS:
        this.signUpMessage = (action.response);
        this.emit('SIGN_UP_SUCCESS');
        break;

      case SIGN_UP_ERROR:
        this.signUpMessage = action.error;
        this.emit('SIGN_UP_ERROR');
        break;

      default:
    }
  }
}

// Initiate an instance of SignInStore
const signUpStore = new SignUpStore();

// Register a dispatcher and bind it to handleActions method
AppDispatcher.register(signUpStore.handleSignUpActions.bind(signUpStore));

// export an instance of SignInStore
export default signUpStore;
