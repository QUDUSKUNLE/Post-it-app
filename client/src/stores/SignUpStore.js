import { EventEmitter } from 'events';
import AppDispatcher from '../dispatcher/AppDispatcher';
import { SIGN_UP_SUCCESS } from '../constants/ActionConstants';

/**
 * Signout Store, it hold user's state, listen to signin Actions
 * @class SignUpStore
 */
class SignUpStore extends EventEmitter {
  /**
   * @constructor
   */
  constructor() {
    super();
    this.signUpMessage = {};
    this.signUpUser = this.signUpUser.bind(this);
    this.handleSignUpActions = this.handleSignUpActions.bind(this);
  }

  /**
   * @description describes a method that saves signUpMessage in the Store
   * @method signUpUser
   * @memberof SignUpStore
   * @returns {object} signUpMessage
   */
  signUpUser() {
    return this.signUpMessage;
  }
  /**
   * Receives actions and update the stores accordingly
   * @method handleSignUpActions
   * @param {object} action - Action payLoad
   * @return {void} -
   */
  handleSignUpActions(action) {
    switch (action.type) {
      case SIGN_UP_SUCCESS:
        this.signUpMessage = action.response;
        this.emit('SIGN_UP_SUCCESS');
        break;

      default:
    }
  }
}

// Initiate an instance of SignInStore
const signUpStore = new SignUpStore();

// Register a dispatcher and bind it to handleActions method
AppDispatcher.register(signUpStore.handleSignUpActions.bind(signUpStore));

// export an instance of SignInStore
export default signUpStore;
