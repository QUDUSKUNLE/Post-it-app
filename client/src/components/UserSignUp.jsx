import React from 'react';
import PropTypes from 'prop-types';
import toastr from 'toastr';
import SignUpStore from '../stores/SignUpStore';
import signUpAction from '../actions/signUpAction';


/**
 * @description - renders SignUp Component
 * @class UserSignUp
 * @extends {React.Component}
 */
export default class UserSignUp extends React.Component {
  /**
   * Create a constructor
   * @constructor
   * @param {object} props -
   */
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      username: '',
      phoneNumber: '',
      confirmPassword: ''
    };

    /**
     * @description This binding is necessary to make `this` work
     * in the callback
     */
    this.onChange = this.onChange.bind(this);
    this.handleSignUpAction = this.handleSignUpAction.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  /**
   * @method componentDidMount
   * @description Adds an event Listener to the Store and fires
   * when the component is fully mounted.
   * @return { void} void
   * @memberof UserSignUp
  */
  componentDidMount() {
    SignUpStore.on('SIGN_UP_SUCCESS', this.handleSignUpAction);
  }

  /**
   * @method componentWillUnmount
   * @description remove event Listener from the Store and fires.
   * @return {void} void
   * @memberof UserSignUp
  */
  componentWillUnmount() {
    SignUpStore.removeListener('SIGN_UP_SUCCESS', this.handleSignUpAction);
  }
  /**
   * @description - onChange event
   * @param {object} event - event.
   * @returns {void} bind input values to name.
   */
  onChange(event) {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  /**
   * @description - this handles SignUp form submission
   * @param {event} event - event.
   * @returns {*} void
  */
  onSubmit(event) {
    event.preventDefault();
    signUpAction({ ...this.state });
  }

  /**
   * handleSignUpAction event
   * @returns {void} .
   */
  handleSignUpAction() {
    const response = SignUpStore.signUpUser();
    toastr.success(response.message);
    this.props.history.push('/signin');
  }

  /**
   * @description - render method, React lifecycle method
   * @returns {Object} SignUp component
   * @SignUp
   */
  render() {
    return (
      <div className="col-md-6">
        <div className="row">
          <div
            className="col-md-10 col-md-offset-1 w3-card w3-white"
            id="signupform"
          >
            <h4>Create an account</h4>
            <br />
            <form onSubmit={this.onSubmit}>
              <div className="form-group">
                <label htmlFor="email">Email</label>
                <input
                  value={this.state.email}
                  onChange={this.onChange}
                  id="email" type="email"
                  className="signinform inp"
                  placeholder="johndoe@example.com"
                  name="email" required
                />
              </div>
              <div className="form-group">
                <label htmlFor="username">Username</label>
                <input
                  value={this.state.username}
                  onChange={this.onChange}
                  id="username"
                  type="text"
                  className="signinform"
                  placeholder="johndoe"
                  name="username" minLength="2" required
                />
              </div>
              <div className="form-group">
                <label htmlFor="phoneNumber">Phone Number</label>
                <input
                  value={this.state.phoneNumber}
                  onChange={this.onChange}
                  id="phoneNumber"
                  type="phone"
                  className="signinform"
                  placeholder="08012345678"
                  name="phoneNumber" minLength="11" required
                />
              </div>
              <div className="form-group">
                <label htmlFor="password">Password</label>
                <input
                  value={this.state.password}
                  onChange={this.onChange}
                  id="pass"
                  type="password"
                  className="signinform"
                  placeholder="********"
                  name="password" minLength="6" required
                />
              </div>
              <div className="form-group">
                <label htmlFor="confirmPassword">Confirm Password</label>
                <input
                  value={this.state.confirmPassword}
                  onChange={this.onChange}
                  id="confirmPassword"
                  type="password"
                  className="signinform"
                  placeholder="********"
                  name="confirmPassword" minLength="6" required
                />
              </div>
              <button type="submit" className="signinformbtn">
                Sign up
              </button>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

// props validation
UserSignUp.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  })
};
