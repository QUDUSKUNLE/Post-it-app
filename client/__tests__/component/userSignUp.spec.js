import React from 'react';
import PropTypes from 'prop-types';
import expect from 'expect';
import { Link } from 'react-router-dom';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';
import signUpResponse from '../../src/__mock__/signUpResponse.json';
import SignUpStore from '../../src/stores/SignUpStore';
import UserSignUp from '../../src/components/UserSignUp';

describe('PostIt-app', () => {
  let wrapper;
  let component;
  const mockOnSignUp = sinon.stub(SignUpStore,
    'on').callsFake((signUp, cb) => cb());
  const mockUnMountSignUp = sinon.stub(SignUpStore,
    'removeListener').callsFake((signUp, cb) => cb());
  const mockSignUpResponse = sinon.stub(SignUpStore,
    'signUpUser').returns('Adekunle');

  beforeEach(() => {
    const props = {
      email: '',
      password: '',
      username: '',
      phoneNumber: '',
      confirmPassword: ''
    };
    component = shallow(<UserSignUp {...props}/>);
    wrapper = mount(<UserSignUp {...props}/>,
      {
        childContextTypes: { router: PropTypes.object },
        context: {
          router:
          {
            history: {
              push: () => null,
              replace: () => null,
              createHref: () => null,
              path: '/',
              component: '[function UserSignUp]',
              location: {
                pathname: '/',
                search: '',
                hash: '',
                key: '6l9jpq'
              },
              computedMatch: {
                path: '/',
                url: '/',
                isExact: true,
                params: {}
              }
            }
          }
        }
      });
  });
  it('expects UserSignUp component to be defined', () => {
    expect(UserSignUp).toBeDefined();
  });
  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
    expect(component.find('div').length).toBe(10);
  });
  // it('should call onChange', () => {
  //   const event = { target: { name: 'name', value: 'value' } };
  //   wrapper.instance().onChange(event);
  //   expect(wrapper.state().name).toEqual('value');
  // });
  // it('should find a link', () => {
  //   console.log(component);
  //   // expect(component.find(Link).at(1).prop('to')).toEqual('/broadcastboard');
  //   // expect(component.find(Link).at(2).prop('to')).toEqual('/member');
  // });
  // it('should call onSubmit', () => {
  //   // wrapper.instance().onSubmit({ preventDefault() {} });
  //   console.log(wrapper.state().email);
  //   // expect(wrapper.state().username).toEqual('');
  //   // expect(wrapper.state().password).toEqual('');
  //   // expect(wrapper.state().confirmPassword).toEqual('');
  //   // expect(wrapper.state().phoneNumber).toEqual('');
  // });
  // it('check if component is being called', () => {
  //   // expect(mockOnSignUp.displayName).toEqual('on');
  //   // console.log(mockOnSignUp.callCount);
  //   // console.log(wrapper.state());
  // });
});
