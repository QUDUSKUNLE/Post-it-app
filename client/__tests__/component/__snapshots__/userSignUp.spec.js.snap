// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostIt-app should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
      className="col-md-5 col-md-offset-1"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12 w3-card w3-white"
          id="signupform"
        >
          <h4>
            Create an account
          </h4>
          <br />
          <form
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="email"
              >
                Email
              </label>
              <input
                className="signinform inp"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="johndoe@example.com"
                required={true}
                type="email"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="username"
              >
                Username
              </label>
              <input
                className="signinform"
                id="username"
                name="username"
                onChange={[Function]}
                placeholder="johndoe"
                required={true}
                type="text"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="phoneNumber"
              >
                Phone Number
              </label>
              <input
                className="signinform"
                id="phoneNumber"
                name="phoneNumber"
                onChange={[Function]}
                placeholder="08012345678"
                required={true}
                type="phone"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="password"
              >
                Password
              </label>
              <input
                className="signinform"
                id="pass"
                name="password"
                onChange={[Function]}
                placeholder="********"
                required={true}
                type="password"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="conf_password"
              >
                Confirm Password
              </label>
              <input
                className="signinform"
                id="confirmPassword"
                name="confirmPassword"
                onChange={[Function]}
                placeholder="********"
                required={true}
                type="password"
                value=""
              />
            </div>
            <button
              className="signinformbtn"
              type="submit"
            >
              Sign up
            </button>
          </form>
          <div
            className="text-center term"
          >
            By clicking "Sign up for Postit App", you agree to our terms of service and privacy policy.
          </div>
        </div>
      </div>
    </div>
  </div>,
  "nodes": Array [
    <div>
      <div
        className="col-md-5 col-md-offset-1"
      >
        <div
          className="row"
        >
          <div
            className="col-md-12 w3-card w3-white"
            id="signupform"
          >
            <h4>
              Create an account
            </h4>
            <br />
            <form
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <label
                  htmlFor="email"
                >
                  Email
                </label>
                <input
                  className="signinform inp"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="johndoe@example.com"
                  required={true}
                  type="email"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="username"
                >
                  Username
                </label>
                <input
                  className="signinform"
                  id="username"
                  name="username"
                  onChange={[Function]}
                  placeholder="johndoe"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="phoneNumber"
                >
                  Phone Number
                </label>
                <input
                  className="signinform"
                  id="phoneNumber"
                  name="phoneNumber"
                  onChange={[Function]}
                  placeholder="08012345678"
                  required={true}
                  type="phone"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="password"
                >
                  Password
                </label>
                <input
                  className="signinform"
                  id="pass"
                  name="password"
                  onChange={[Function]}
                  placeholder="********"
                  required={true}
                  type="password"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="conf_password"
                >
                  Confirm Password
                </label>
                <input
                  className="signinform"
                  id="confirmPassword"
                  name="confirmPassword"
                  onChange={[Function]}
                  placeholder="********"
                  required={true}
                  type="password"
                  value=""
                />
              </div>
              <button
                className="signinformbtn"
                type="submit"
              >
                Sign up
              </button>
            </form>
            <div
              className="text-center term"
            >
              By clicking "Sign up for Postit App", you agree to our terms of service and privacy policy.
            </div>
          </div>
        </div>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <UserSignUp
        confirmPassword=""
        email=""
        password=""
        phoneNumber=""
        username=""
      />,
      "_debugID": 29,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": UserSignUp {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleSignUpAction": [Function],
        "onChange": [Function],
        "onSubmit": [Function],
        "props": Object {
          "confirmPassword": "",
          "email": "",
          "password": "",
          "phoneNumber": "",
          "username": "",
        },
        "refs": Object {},
        "state": Object {
          "confirmPassword": "",
          "email": "",
          "password": "",
          "phoneNumber": "",
          "username": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
            className="col-md-5 col-md-offset-1"
          >
            <div
              className="row"
            >
              <div
                className="col-md-12 w3-card w3-white"
                id="signupform"
              >
                <h4>
                  Create an account
                </h4>
                <br />
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="email"
                    >
                      Email
                    </label>
                    <input
                      className="signinform inp"
                      id="email"
                      name="email"
                      onChange={[Function]}
                      placeholder="johndoe@example.com"
                      required={true}
                      type="email"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="username"
                    >
                      Username
                    </label>
                    <input
                      className="signinform"
                      id="username"
                      name="username"
                      onChange={[Function]}
                      placeholder="johndoe"
                      required={true}
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="phoneNumber"
                    >
                      Phone Number
                    </label>
                    <input
                      className="signinform"
                      id="phoneNumber"
                      name="phoneNumber"
                      onChange={[Function]}
                      placeholder="08012345678"
                      required={true}
                      type="phone"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="password"
                    >
                      Password
                    </label>
                    <input
                      className="signinform"
                      id="pass"
                      name="password"
                      onChange={[Function]}
                      placeholder="********"
                      required={true}
                      type="password"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="conf_password"
                    >
                      Confirm Password
                    </label>
                    <input
                      className="signinform"
                      id="confirmPassword"
                      name="confirmPassword"
                      onChange={[Function]}
                      placeholder="********"
                      required={true}
                      type="password"
                      value=""
                    />
                  </div>
                  <button
                    className="signinformbtn"
                    type="submit"
                  >
                    Sign up
                  </button>
                </form>
                <div
                  className="text-center term"
                >
                  By clicking "Sign up for Postit App", you agree to our terms of service and privacy policy.
                </div>
              </div>
            </div>
          </div>
        </div>,
        "_debugID": 30,
        "_renderedOutput": <div>
          <div
            className="col-md-5 col-md-offset-1"
          >
            <div
              className="row"
            >
              <div
                className="col-md-12 w3-card w3-white"
                id="signupform"
              >
                <h4>
                  Create an account
                </h4>
                <br />
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="email"
                    >
                      Email
                    </label>
                    <input
                      className="signinform inp"
                      id="email"
                      name="email"
                      onChange={[Function]}
                      placeholder="johndoe@example.com"
                      required={true}
                      type="email"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="username"
                    >
                      Username
                    </label>
                    <input
                      className="signinform"
                      id="username"
                      name="username"
                      onChange={[Function]}
                      placeholder="johndoe"
                      required={true}
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="phoneNumber"
                    >
                      Phone Number
                    </label>
                    <input
                      className="signinform"
                      id="phoneNumber"
                      name="phoneNumber"
                      onChange={[Function]}
                      placeholder="08012345678"
                      required={true}
                      type="phone"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="password"
                    >
                      Password
                    </label>
                    <input
                      className="signinform"
                      id="pass"
                      name="password"
                      onChange={[Function]}
                      placeholder="********"
                      required={true}
                      type="password"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="conf_password"
                    >
                      Confirm Password
                    </label>
                    <input
                      className="signinform"
                      id="confirmPassword"
                      name="confirmPassword"
                      onChange={[Function]}
                      placeholder="********"
                      required={true}
                      type="password"
                      value=""
                    />
                  </div>
                  <button
                    className="signinformbtn"
                    type="submit"
                  >
                    Sign up
                  </button>
                </form>
                <div
                  className="text-center term"
                >
                  By clicking "Sign up for Postit App", you agree to our terms of service and privacy policy.
                </div>
              </div>
            </div>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <UserSignUp
    confirmPassword=""
    email=""
    password=""
    phoneNumber=""
    username=""
  />,
}
`;
